multicluster/controllers/multicluster/member_clusterset_controller.go

```diff
@@ -310,38 +315,62 @@ func (r *MemberClusterSetReconciler) updateStatus() {
 	clusterSet.Status = status
 	err = r.Status().Update(context.TODO(), clusterSet)
 	if err != nil {
-		klog.ErrorS(err, "Failed to update Status of ClusterSet", "Name", namespacedName)
+		klog.ErrorS(err, "Failed to update Status of ClusterSet", "name", namespacedName)
 	}
 }
 
-// SetRemoteCommonAreaManager is for testing only
-func (r *MemberClusterSetReconciler) SetRemoteCommonAreaManager(mgr commonarea.RemoteCommonAreaManager) commonarea.RemoteCommonAreaManager {
-	r.remoteCommonAreaManager = mgr
-	return r.remoteCommonAreaManager
+// SetRemoteCommonArea is for testing only
+func (r *MemberClusterSetReconciler) SetRemoteCommonArea(commanArea commonarea.RemoteCommonArea) commonarea.RemoteCommonArea {
+	r.remoteCommonArea = commanArea
+	return r.remoteCommonArea
 }
 
 func (r *MemberClusterSetReconciler) GetRemoteCommonAreaAndLocalID() (commonarea.RemoteCommonArea, string, error) {
 	r.mutex.RLock()
 	defer r.mutex.RUnlock()
-	var remoteCommonArea commonarea.RemoteCommonArea
-	if r.remoteCommonAreaManager == nil {
+	if r.remoteCommonArea == nil {
 		return nil, "", errors.New("ClusterSet has not been initialized, no available Common Area")
 	}
 
-	remoteCommonAreas := r.remoteCommonAreaManager.GetRemoteCommonAreas()
-	if len(remoteCommonAreas) <= 0 {
-		return nil, "", errors.New("ClusterSet has not been set up properly, no available Common Area")
+	if r.remoteCommonArea.IsConnected() {
+		localClusterID := string(r.remoteCommonArea.GetLocalClusterID())
+		return r.remoteCommonArea, localClusterID, nil
+	}
+	return nil, "", errors.New("no connected remote common area")
+}
+
+func (r *MemberClusterSetReconciler) checkLeaderConnection() {
```

Comment:
Should we implement this func in remoteCommonArea itself (if it is really needed)?